### ISSUE TITLE
Cannot reuse existing API Gateway custom domain

### ISSUE THREAD

**Issue by gmunguia** [serverless-stack/serverless-stack#226]
First things first: thanks for creating this library and open sourcing it 🤗.

According to [the example in the documentation](https://docs.serverless-stack.com/constructs/Api#reusing-an-existing-api-gateway-custom-domain), it is possible to do the following:

```typescript
new Api(this, "Api", {
  customDomain: {
    domainName: apigatewayv2.DomainName.fromDomainNameAttributes(
      this,
      "MyDomain",
      {
        name,
        regionalDomainName,
        regionalHostedZoneId,
      }
    ),
    path: "newPath",
  },
  routes: {
    "GET /notes": "src/list.main",
  },
});
```

However, that causes a type error, because [`domainName` expects `string | DomainName`](https://github.com/serverless-stack/serverless-stack/blob/master/packages/resources/src/Api.ts#L71), but [`fromDomainNameAttributes` returns an `IDomainName`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-apigatewayv2.DomainName.html#static-fromwbrdomainwbrnamewbrattributesscope-id-attrsspan-classapi-icon-api-icon-experimental-titlethis-api-element-is-experimental-it-may-change-without-noticespan).

Having a look at the source code, [`domainName` is passed to `HttpApi` constructor as part of `defaultDomainMapping`](https://github.com/serverless-stack/serverless-stack/blob/master/packages/resources/src/Api.ts#L160), [which expects an `IDomainName`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-apigatewayv2.DomainMappingOptions.html). So it would be safe to change the typing of `domainName` to `IDomainName`.

Happy to make a PR for this if help is needed.

**Comment by fwang** [comment]
@gmunguia Nice catch! A PR would be great!

**Comment by fwang** [comment]
Thanks @gmunguia The fix has been released in v0.9.17
